Providers/Scripts/2.4x-2.5x/Scripts/nxOMSAgentNPMConfig.py

```diff
@@ -0,0 +1,207 @@
+    if Contents is not None:
+        Contents = base64.b64decode(Contents)#Contents.encode('ascii', 'ignore')
+    else:
+        Contents = ''
+    if Ensure is not None and Ensure != '':
+        Ensure = Ensure.encode('ascii', 'ignore')
+    else:
+        Ensure = 'Present'
+    if ContentChecksum is not None:
+        ContentChecksum = ContentChecksum.encode('ascii', 'ignore')
+    else:
+        ContentChecksum = ''
+    return ConfigType, ConfigID, Contents, Ensure, ContentChecksum
+
+
+def Set_Marshall(ConfigType, ConfigID, Contents, Ensure, ContentChecksum):
+    (ConfigType, ConfigID, Contents, Ensure, ContentChecksum) = init_vars(ConfigType, ConfigID, Contents, Ensure, ContentChecksum)
+    retval = Set(ConfigType, ConfigID, Contents, Ensure, ContentChecksum)
+    return retval
+
+
+def Test_Marshall(ConfigType, ConfigID, Contents, Ensure, ContentChecksum):
+    (ConfigType, ConfigID, Contents, Ensure, ContentChecksum) = init_vars(ConfigType, ConfigID, Contents, Ensure, ContentChecksum)
+    retval = Test(ConfigType, ConfigID, Contents, Ensure, ContentChecksum)
+    return retval
+
+
+def Get_Marshall(ConfigType, ConfigID, Contents, Ensure, ContentChecksum):
+    arg_names = list(locals().keys())
+    (ConfigType, ConfigID, Contents, Ensure, ContentChecksum) = init_vars(ConfigType, ConfigID, Contents, Ensure, ContentChecksum)
+    retval = 0
+    retval = Get(ConfigType, ConfigID, Contents, Ensure, ContentChecksum)
+    ConfigType = protocol.MI_String(ConfigType)
+    ConfigID = protocol.MI_String(ConfigID)
+    Ensure = protocol.MI_String(Ensure)
+    Contents = protocol.MI_String(Contents)
+    ContentChecksum = protocol.MI_String(ContentChecksum)
+
+    retd = {}
+    ld = locals()
+    for k in arg_names:
+        retd[k] = ld[k]
+    return retval, retd
+
+
+############################################################
+# Begin user defined DSC functions
+############################################################
+
+def SetShowMof(a):
+    global show_mof
+    show_mof = a
+
+
+def ShowMof(op, ConfigType, ConfigID, Contents, Ensure, ContentChecksum):
+    if not show_mof:
+        return
+    mof = ''
+    mof += op + ' nxOMSAgentNPMConfig MyNPMConfig \n'
+    mof += '{\n'
+    mof += '    ConfigType = "' + ConfigType + '"\n'
+    mof += '    ConfigID = "' + ConfigID + '"\n'
+    mof += '    Contents = "' + Contents + '"\n'
+    mof += '    Ensure = "' + Ensure + '"\n'
+    mof += '    ContentChecksum = "' + ContentChecksum + '"\n'
+    mof += '}\n'
+    f = open('./test_mofs.log', 'a')
+    Print(mof, file=f)
+    LG().Log('INFO', mof)
+    f.close()
+
+
+def Set(ConfigType, ConfigID, Contents, Ensure, ContentChecksum):
+    ShowMof('SET', ConfigType, ConfigID, Contents, Ensure, ContentChecksum)
+    retval = 0
+    destFileFullPath = CONFIG_PATH.__add__(DEST_FILE_NAME)
+    if ConfigType != 'UpdateNPMAgentConfig' or Ensure != 'Present':
+        LG().Log('ERROR', 'Config type or ensure did not match, exiting set')
```

Comment:
why raise the error when Ensure is not "Present". I think it should disable the NPM agent when it's "Absent"